include/dxc/dxcapi.h

```diff
@@ -145,153 +154,253 @@ typedef struct DxcShaderHash {
+/// output blobs, such as errors/warnings, preprocessed HLSL, or other text are
+/// returned using encodings based on the -encoding option passed to the
+/// compiler.
 struct IDxcBlobWide : public IDxcBlobEncoding {
 public:
+  /// \brief Retrieves a pointer to the string stored in this blob.
   virtual LPCWSTR STDMETHODCALLTYPE GetStringPointer(void) = 0;
+
+  /// \brief Retrieves the length of the string stored in this blob, in
+  /// characters, excluding the null-terminator.
   virtual SIZE_T STDMETHODCALLTYPE GetStringLength(void) = 0;
 };
+
 CROSS_PLATFORM_UUIDOF(IDxcBlobUtf8, "3DA636C9-BA71-4024-A301-30CBF125305B")
+/// \brief A blob containing a UTF-8 encoded string
+///
+/// The value returned by GetBufferSize() is the size of the buffer, in bytes,
+/// including the null-terminator.
+///
+/// Depending on the -encoding option passed to the compiler, this interface is
+/// used to return string output blobs, such as errors/warnings, preprocessed
+/// HLSL, or other text. Output name strings always use IDxcBlobWide.
 struct IDxcBlobUtf8 : public IDxcBlobEncoding {
 public:
+  /// \brief Retrieves a pointer to the string stored in this blob.
   virtual LPCSTR STDMETHODCALLTYPE GetStringPointer(void) = 0;
+
+  /// \brief Retrieves the length of the string stored in this blob, in
+  /// characters, excluding the null-terminator.
   virtual SIZE_T STDMETHODCALLTYPE GetStringLength(void) = 0;
 };
 
-// Define legacy name IDxcBlobUtf16 as IDxcBlobWide for Win32
 #ifdef _WIN32
+/// IDxcBlobUtf16 is a legacy alias for IDxcBlobWide on Win32
 typedef IDxcBlobWide IDxcBlobUtf16;
 #endif
 
 CROSS_PLATFORM_UUIDOF(IDxcIncludeHandler, "7f61fc7d-950d-467f-b3e3-3c02fb49187c")
+/// \brief Interface for handling include directives
+///
+/// This interface can be implemented to customize handling of include
+/// directives.
+///
+/// Use IDxcUtils::CreateDefaultIncludeHandler to create a default
+/// implementation that reads include files from the filesystem..
+///
 struct IDxcIncludeHandler : public IUnknown {
+  /// \brief Load a source file to be included by the compiler.
+  ///
+  /// \param pFilename candidate filename.
+  ///
+  /// \param ppIncludeSource Resultant source object for included file, nullptr
+  /// if not found.
   virtual HRESULT STDMETHODCALLTYPE LoadSource(
-    _In_z_ LPCWSTR pFilename,                                 // Candidate filename.
-    _COM_Outptr_result_maybenull_ IDxcBlob **ppIncludeSource  // Resultant source object for included file, nullptr if not found.
+      _In_z_ LPCWSTR pFilename,
+      _COM_Outptr_result_maybenull_ IDxcBlob **ppIncludeSource 
     ) = 0;
 };
 
-// Structure for supplying bytes or text input to Dxc APIs.
-// Use Encoding = 0 for non-text bytes, ANSI text, or unknown with BOM.
+/// \brief Structure for supplying bytes or text input to Dxc APIs.
 typedef struct DxcBuffer {
+  /// \brief pointer to the start of the buffer
   LPCVOID Ptr;
+
+  /// \brief size of the buffer, in bytes
```

Comment:
nit: maybe `Size of the buffer (in bytes).` or `Size of the buffer in bytes.` I don't think the comma is correct.